File: Ecommerce Java/Product.java
public class Product {
    String name;
    int price;
    int quantity;
    ProductDetails details;

    Product(String p_name, int p_price, int p_quantity, ProductDetails details){
        name = p_name;
        price = p_price;
        quantity = p_quantity;
        this.details = details;
    }
    public String getName(){
        return this.name;
    }
    public int getPrice(){
        return this.price;
    }
    public int getQuantity(){
        return this.quantity;
    }
    public String getProductDetail(){
        return this.details.getDescription();
    }
}

File: scalaproject/src/main/scala/Product.scala
class Product(var p_name:String, var p_price:Double, var p_quantity:Int)
{
    def getName(): String =
    {
        p_name
    }
    
    def setName(newName:String): Unit =
    {
        p_name = newName
    }

    def getPrice(): Double =
    {
        p_price
    }

    def setPrice(newPrice:Double): Unit =
    {
        p_price = newPrice
    }

    def getQuantity(): Int =
    {
        p_quantity
    }

    def setQuantity(newQuantity:Int): Unit =
    {
        p_quantity = newQuantity
    }
}